db/dbScript.sql

```diff
@@ -0,0 +1,393 @@
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),
+  requested_at TIMESTAMP NOT null check (requested_at <= NOW()), --time of the request
+  origin_latitude FLOAT NOT null check (origin_latitude between -90 and 90), --latitude of the starting location
+  origin_longitude FLOAT NOT null check (origin_longitude between -180 and 180),
+  departure_time TIME, --preferred departure_time specified by user
+  weekday INT check (weekday between 0 and 6), --day of the week for which the user wants to receive the recommendations 
+  arrival_time TIME check (arrival_time > departure_time), --preferred arrival time specified by user
+  time_priority INT NOT null check (time_priority between 0 and 100), --the value of the travelling time slider on the frontend
+  tcost_priority INT NOT null check (tcost_priority between 0 and 100 and tcost_priority + time_priority = 100),
+  min_congestion_rating NUMERIC(4,2) NOT null check (min_congestion_rating between 1 and 5),
+  min_rating NUMERIC(4,2) NOT null check (min_rating between 1 and 5),
+  min_membership_price INT NOT null check (min_membership_price >= 0),
+  name VARCHAR(50),--each user can specify the name of the request 
+  user_id uuid NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (user_id) REFERENCES Account(id),
+  UNIQUE (user_id, name),
+  UNIQUE (user_id, requested_at)
+);
+create index if not exists idx_request_user_id on Request(user_id);
+create index if not exists idx_request_name on Request("name");
+
+CREATE TABLE if not exists Recommendation
+(
+  tcost NUMERIC(4,2) NOT null check (tcost >= 0), --total travelling cost to get to the gym
+  time TIME NOT NULL, --total travelling time to the gym
+  time_score NUMERIC(4,2) NOT null check (time_score between 0 and 10), --calculated score for the travelling time
+  tcost_score NUMERIC(4,2) NOT null check (tcost_score between 0 and 10), --calculated score for the total price
+  congestion_score NUMERIC(4,2) check (congestion_score between 0 and 10),
+  rating_score NUMERIC(4,2) check (rating_score between 0 and 10),
+  total_score NUMERIC(4,2) NOT null check (total_score between 0 and 10),
+  type rec_type NOT NULL, --main or alternative rating
+  gym_id uuid NOT NULL,
+  request_id uuid NOT NULL,
+  currency_id uuid NOT NULL,
+  PRIMARY KEY (gym_id, request_id),
+  FOREIGN KEY (gym_id) REFERENCES Gym(id),
+  FOREIGN KEY (request_id) REFERENCES Request(id),
+  FOREIGN KEY (currency_id) REFERENCES Currency(id)
+);
+create index if not exists idx_recommendation_request_id on Recommendation(request_id);
+
+CREATE TABLE if not exists Rating
+(
+  created_at TIMESTAMP NOT null check (created_at <= NOW()),
+  rating INT NOT null check (rating between 0 and 5),
+  user_id uuid NOT NULL,
+  gym_id uuid NOT NULL,
+  PRIMARY KEY (user_id, gym_id),
+  FOREIGN KEY (user_id) REFERENCES Account(id),
+  FOREIGN KEY (gym_id) REFERENCES Gym(id)
+);
+create index if not exists idx_rating_gym_id on Rating(gym_id);
+
+CREATE TABLE if not exists Congestion_rating
```

Comment:
What is the purpose of this table and how **congestion_rating** is going to be calculated?